shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/plants/data/FilePlantsDataSource.kt

```diff
@@ -1,17 +1,62 @@
 package com.vml.tutorial.plantshop.plants.data
 
+import com.vml.tutorial.plantshop.MR.files.plants
+import com.vml.tutorial.plantshop.PlantDatabase
+import com.vml.tutorial.plantshop.core.data.FileReader
+import com.vml.tutorial.plantshop.core.data.toPlant
 import com.vml.tutorial.plantshop.core.data.toPlants
 import com.vml.tutorial.plantshop.plants.domain.Plant
 import com.vml.tutorial.plantshop.plants.domain.PlantsDataSource
-import com.vml.tutorial.plantshop.MR.files.plants
-import com.vml.tutorial.plantshop.core.data.FileReader
-
 
 class FilePlantsDataSource(
-    private val fileReader: FileReader
-): PlantsDataSource {
+    private val fileReader: FileReader,
+    db: PlantDatabase
+) : PlantsDataSource {
+    private val queries = db.plantQueries
+
     override fun getPlants(): List<Plant> {
-        val rawData: String = fileReader.loadFile(plants)?: return emptyList()
-        return rawData.toPlants()
+        val rawData: String = fileReader.loadFile(plants) ?: return emptyList()
+        return getPlantsWithFavoriteState(getFavorites(), rawData.toPlants())
+    }
+
+    override fun getFavorites(): List<Plant> {
+        return queries.getFavoritePlants().executeAsList().map { plantEntity ->
+            plantEntity.toPlant()
+        }
+    }
+
+    private fun getPlantsWithFavoriteState(
+        favorites: List<Plant>,
+        plants: List<Plant>
+    ): List<Plant> {
+        for (favorite in favorites) {
+            val matchingPlant = plants.find { it.id == favorite.id }
+            matchingPlant?.isFavorite = true
+        }
+        return plants
```

Comment:
You can simplify it -> plants.map { plant -> plant.copy(isFavorite = favorites.any { it.id == plant.id }) }